src/util/sourceMappingDecoder.js

```diff
@@ -0,0 +1,132 @@
+'use strict'
+var util = require('../helpers/util')
+
+/*
+  Decompress the source mapping given by solc-bin.js
+*/
+function SourceMappingDecoder () {
+  // s:l:f:j
+}
+
+/*
+    * Decode the source mapping for the given @arg index
+    *
+    * @param {Integer} index      - source mapping index to decode
+    * @param {String} mapping     - compressed source mapping given by solc-bin
+    * @param {Function} callback  - returns the decompressed source mapping for the given index {start, length, file, jump}
+*/
+SourceMappingDecoder.prototype.decompress = function (index, mapping, cb) {
+  var ret = {}
+  var map = mapping.split(';')
+  for (var k = index; k >= 0; k--) {
+    var current = map[k]
+    if (!current.length) {
+      continue
+    }
+    current = current.split(':')
+    if (ret.start === undefined && current[0] && current[0].length) {
+      ret.start = parseInt(current[0])
+    }
+    if (ret.length === undefined && current[1] && current[1].length) {
+      ret.length = parseInt(current[1])
+    }
+    if (ret.file === undefined && current[2] && current[2].length) {
+      ret.file = parseInt(current[2])
+    }
+    if (ret.jump === undefined && current[3] && current[3].length) {
+      ret.jump = current[3]
+    }
+    if (ret.start !== undefined && ret.length !== undefined && ret.file !== undefined && ret.jump !== undefined) {
+      break
+    }
+  }
+  cb(null, ret)
+}
+
+/*
+    * Decode the source mapping for the given compressed mapping
+    *
+    * @param {String} mapping     - compressed source mapping given by solc-bin
+    * @param {Function} callback  - returns the decompressed source mapping. Array of {start, length, file, jump}
+*/
+SourceMappingDecoder.prototype.decompressAll = function (mapping, cb) {
+  var map = mapping.split(';')
+  var ret = []
+  for (var k in map) {
+    var compressed = map[k].split(':')
+    var sourceMap = {
+      start: compressed[0] ? parseInt(compressed[0]) : ret[ret.length - 1].start,
+      length: compressed[1] ? parseInt(compressed[1]) : ret[ret.length - 1].length,
+      file: compressed[2] ? parseInt(compressed[2]) : ret[ret.length - 1].file,
+      jump: compressed[3] ? compressed[3] : ret[ret.length - 1].jump
+    }
+    ret.push(sourceMap)
+  }
+  cb(null, ret)
+}
+
+/*
+    * Retrieve line/column position of each source char
+    *
+    * @param {String} source - contract source code
+    * @param {Function} callback - returns an array containing line break char position for each line
```

Comment:
No callback, just returns the offsets of the line breaks (and also not "for each line").